### PR TITLE
Changes the item price to 1 if it is 0

### PR DESCRIPTION
**Pull request by dastgirp**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

If Item Price is 0, it is now changed to 1.

[//]: # (Describe at length, the changes that this pull request makes.)

**Affected Branches:** 

[//]: # (Master? Slave?)

**Issues addressed:**

#2177 

[//]: # (Issue Tracker Number if any.)

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2208)

File: src/map/pc.c [hunk 1]
```diff
@@ -4403,7 +4403,7 @@ static int pc_modifybuyvalue(struct map_session_data *sd, int orig_value)
 {
 	int skill_lv, rate1 = 0, rate2 = 0;
 	if (orig_value <= 0)
-		return 0;
+		return 1;
```
Comment (4144) [inline src/map/pc.c]: probably better add battle config option?
because price 0 i think make sense. but by default better follow official way with 1 price

Comment (dastgirp) [reply src/map/pc.c]: Fixed

File: src/map/battle.h [hunk 2]
```diff
@@ -570,6 +570,8 @@ struct Battle_Config {
 
 	int feature_enable_attendance_system;
 	int feature_attendance_endtime;
+
+	int min_item_buy_price, min_item_sell_price;
```
Comment (4144) [inline src/map/battle.h]: better put each variable in own line

File: conf/map/battle/items.conf [hunk 3]
```diff
@@ -123,3 +123,8 @@ boarding_halter_speed: 25
 // Allow to use items when the storage is open?
 // Official: false (Default)
 storage_use_item: false
+
+// Minimum item buy/sell price at shop
+// Default: 1
+min_item_buy_price: 1
+min_item_sell_price: 1
```
Comment (kyeme) [inline conf/map/battle/items.conf]: On official the sell price is still 0z

Comment (dastgirp) [reply conf/map/battle/items.conf]: done

File: src/map/battle.h (hunk 2 above)
Comment (dastgirp) [reply src/map/battle.h]: done

**Comment by dastgirp** [comment]
@kyeme @4144 Please check again

**Comment by kyeme** [comment]
Looks good to me.

**Comment by skyleo** [comment]
I don't recommend changing this default setting, since a lot of WoE Player and other kind of stuff use this feature to exchange WoE Materials to their mates quickly.

**Pull request merged**